pkg/plugin/plugin.go

```diff
@@ -169,3 +171,71 @@ const (
 	ValidationTypeExclusion
 	ValidationTypeRegex
 )
+
+const (
+	PluginTypeBuiltin    = "builtin"
+	PluginTypeStandalone = "standalone"
+	PluginTypeAny        = "any"
+
+	PluginVersionLatest = "latest"
+)
+
+type FullName string
```

Comment:
I don't see how this would make it better. If you are saying we wouldn't need a 2d map in the registries, I have to say I did it on purpose that way, we can provide better error messages like "_couldn't find plugin X:v0.1.0, available versions are [v0.2.0 v0.3.4]_". At least that was the idea behind that. If we had a 1d map we would have to loop through all keys to figure out what other versions of a plugin we have.

Reply:
> There would be no need to parse the full name string when getting the type, version or name.

You would have to parse it when creating the struct, now we just do it lazily when we need it.

Reply:
Ah, I see, the name from the configs is parsed into FullName. Personally, I'd still parse it once (and also early on make sure it's a valid name to begin with), then parse it whenever type/name/version are needed (it's also less code). But I'd still rate this as a small thing, not a PR blocker.